README.md

```diff
@@ -62,16 +26,74 @@ These are the currently supported instrumentations with the minimum version requ
-we vendor this version of the library inside the integration. Under normal circumstances this is not something that we want to do, because users using this integration
-might be running versions different from the vendored one, creating hard to debug conflicts.
+## Configuration values
 
-To run integration tests locally, you should set the `INTEGRATION` environment variable. The dependencies of the integration tests are best run via Docker. To get an
-idea about the versions and the set-up take a look at our [CI config](https://github.com/signalfx/signalfx-go-tracing/blob/master/.circleci/config.yml).
+Set configuration values from environment variables or directly in your code:
 
-The best way to run the entire test suite is using the [CircleCI CLI](https://circleci.com/docs/2.0/local-jobs/). Simply run `circleci build`
-in the repository root. Note that you might have to increase the resources dedicated to Docker to around 4GB.
+| Code | Environment Variable | Default Value | Notes |
+| ---  | ---                  | ---           | ---   |
+| [WithServiceName](https://godoc.org/github.com/signalfx/signalfx-go-tracing/tracing/#WithServiceName) | `SIGNALFX_SERVICE_NAME` | `SignalFx-Tracing` | The name of the service. |
+| [WithEndpointURL](https://godoc.org/github.com/signalfx/signalfx-go-tracing/tracing/#WithEndpointURL) | `SIGNALFX_ENDPOINT_URL` | `http://localhost:9080/v1/trace` | The URL to send traces to.  |
+| [WithAccessToken](https://godoc.org/github.com/signalfx/signalfx-go-tracing/tracing/#WithAccessToken) | `SIGNALFX_ACCESS_TOKEN` | none | The access token for your SignalFx organization. |
+| [WithGlobalTag](https://godoc.org/github.com/signalfx/signalfx-go-tracing/tracing/#WithGlobalTag) | `SIGNALFX_SPAN_TAGS` | none | Comma-separated list of tags included in every reported span. For example, "key1:val1,key2:val2". Use only string values for tags.|
+
+## Instrument a Go application
+
+Follow these steps to instrument target libraries with provided wrapper libraries. 
+
+For more information about how to instrument a Go application, see the
+[examples](https://github.com/signalfx/tracing-examples/tree/master/signalfx-tracing/signalfx-go-tracing).
+
+1. Add `github.com/signalfx/signalfx-go-tracing` to your `go mod` or `dep`
+dependencies.
+2. Specify the name of the service you're instrumenting:
+   ```bash
+    $ export SIGNALFX_SERVICE_NAME = "your_service"
+   ```
+3. If you're sending trace data to an endpoint different than `http://localhost:9080/v1/trace`,
+set the endpoint URL:
+   ```bash
+   $ export SIGNALFX_ENDPOINT_URL = "your_endpoint"
+   ```
+4. If you're sending trace data directly to a SignalFx ingest endpoint or to an
+OpenTelemetry Collector, set the access token for your SignalFx organization:
+   ```bash
+   $ export SIGNALFX_ACCESS_TOKEN = "your_access_token"
+   ```
+5. Import the wrapper library for the target library you want to instrument and
```

Comment:
Not all instrumentations are wrappers. Some are middlewares for example. May be call these instrumentations or intrumentors?